### ISSUE TITLE
Error after importing @angular/material": "^2.0.0-beta.8"

### ISSUE THREAD

**Issue by ravivit9** [mgechev/angular-seed#2031]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see use [gitter](https://gitter.im/mgechev/angular2-seed) or [stackoverflow](https://stackoverflow.com/questions/tagged/angular2)
```

**Current behavior**
After downloading the seed it worked fine.  When installed material design beta8 version I started getting this error

**Expected behavior**
Runtime error as below screenshot.

**Minimal reproduction of the problem with instructions**
Import material2 design in home.module.ts to get this error.

**What is the motivation / use case for changing the behavior?**
To make use of material2 design with angular-seed.

**Please tell us about your environment:**
Mac, npm, Webstorm.

* **Angular Seed Version:** `aaaaf75`
angular-seed (downloaded today 23-jul-2017)
Last commit: facd075 / facd07577735ba61b50c2e7443db18415175aa04

Node: v6.11.0
<img width="888" alt="screen shot 2017-07-23 at 19 16 16" src="https://user-images.githubusercontent.com/3435829/28501798-6fb15e60-6fdb-11e7-8bb8-2367e82a048e.png">

Added the following on top of existing angular-seed code.  No change to the code done yet.

app.module.ts
===========
`import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
`............
............

home.component.html
=================
`<md-card>
  <md-slide-toggle
    class="example-margin">
    Slide me!
  </md-slide-toggle>
</md-card>
`............
............

home.module.ts
===========
`import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import {  MdSliderModule, MdSlideToggleModule } from '@angular/material';`
............
............

`@NgModule({
  imports: [ HomeRoutingModule, SharedModule,
    BrowserModule,
    BrowserAnimationsModule,
    FormsModule,
    //MdNativeDateModule,
    ReactiveFormsModule,
  ],
  exports: [HomeComponent, MdSliderModule, MdSlideToggleModule]
  bootstrap: [HomeComponent],
  providers: [NameListService]
})

export class HomeModule { }
platformBrowserDynamic().bootstrapModule(HomeModule);`

**Comment by adibb83** [comment]
same for me

**Comment by ravivit9** [comment]
downloaded ng4-material seed which worked well along with angular but may not have all features as equivalent to angular-seed repo.  I really like this repo as it's well maintained to recent.

Just create >>> ng new ng4-material and follow the instruction from the link.  Config wise not much change I found between my project and ng4-material project.
https://coursetro.com/posts/code/67/Angular-4-Material-Tutorial

**Comment by zladuric** [comment]
@ravivit9  Did you add the link to umd (or whatever) in project.config?

Take a look at: https://github.com/mgechev/angular-seed/wiki/Add-external-dependency and https://github.com/mgechev/angular-seed/wiki/Add-external-scripts-and-styles

**Comment by vyakymenko** [comment]
Hi @ravivit9 ,

Let's try step-by-step:

1. Install main packages:
```sh
$  npm install @angular/material @angular/cdk material-design-lite hammerjs --save
```

2. Configure `project.config.ts`:
```ts
...
import { ExtendPackages } from './seed.config.interfaces';

...

    // Add `NPM` third-party libraries to be injected/bundled.
    this.NPM_DEPENDENCIES = [
      ...this.NPM_DEPENDENCIES,
      {src: 'material-design-lite/dist/material.js', inject: 'libs'}
      // {src: 'jquery/dist/jquery.min.js', inject: 'libs'},
      // {src: 'lodash/lodash.min.js', inject: 'libs'},
    ];
.....

    // Add packages (e.g. ng2-translate)
    let additionalPackages: ExtendPackages[] = [
      {
        name: '@angular/material',
        path: 'node_modules/@angular/material/bundles/material.umd.js'
      },{
        name: '@angular/cdk',
        path: 'node_modules/@angular/cdk/bundles/cdk.umd.js'
      },{
        name: 'hammerjs',
        path: 'node_modules/hammerjs/hammer.js'
      }
    ];

    this.addPackagesBundles(additionalPackages);
....
}
```

3. Configure `app.module.ts`:

```ts
...
import 'hammerjs';
```

4. Configure `shared.module.ts`:

```ts
...

import { MdButtonModule } from '@angular/material';
...

@NgModule({
  imports: [CommonModule, RouterModule,
    MdButtonModule],
  declarations: [ToolbarComponent, NavbarComponent],
  exports: [ToolbarComponent, NavbarComponent,
    CommonModule, FormsModule, RouterModule,
    MdButtonModule]
})
...
```

5. Configure `index.html` for example:
```html
...
<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
  <link rel="stylesheet" href="https://code.getmdl.io/1.3.0/material.blue_grey-light_green.min.css" />
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700" type="text/css">
...

```

6. Last thing, we need to verify everything `home.component.html`:

```html
...
<!--<button type="submit">Add</button>-->
  <button md-button type="submit">
    Add
  </button>
....

```

**Comment by vyakymenko** [comment]
@mgechev , I think we can close this issue.

**Comment by ravivit9** [comment]
just implemented as guided above.  Got an error after running
The selector "sd-home" did not match any elements
<img width="988" alt="screen shot 2017-07-24 at 11 11 16" src="https://user-images.githubusercontent.com/3435829/28518732-d875f2fa-7060-11e7-8533-7d40fd879e87.png">

**Comment by ravivit9** [comment]
BTW, thanks for the step by step guidance.  Once the above error sorted out then will be good to go.

**Comment by vyakymenko** [comment]
@ravivit9 , show me your index html page `body`.

**Comment by ravivit9** [comment]
`<body>

  <sd-app>Loading...</sd-app>

  <script defer>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', '<%= GOOGLE_ANALYTICS_ID %>', 'auto');
    ga('send', 'pageview');
  </script>

  <script>
  // Fixes undefined module function in SystemJS bundle
  function module() {}
  </script>

  
  

  <% if (BUILD_TYPE === 'dev') { %>
  <script src="<%= APP_BASE %>system-config.js"></script>
  <% } %>

  
  

  
  

  <% if (BUILD_TYPE === 'dev') { %>
  <script>
  System.import('<%= BOOTSTRAP_MODULE %>')
    .catch(function (e) {
      console.error(e.stack || e,
        'Not expecting this error? Report it at https://github.com/mgechev/angular-seed/issues');
    });
  </script>
  <% } %>

</body>`

**Comment by vyakymenko** [comment]
@ravivit9 , what tag is used in `Loading...`. Please show code, not plane text.

**Comment by ravivit9** [comment]
It's from app.component.ts file
`import { Component } from '@angular/core';
import { Config } from './shared/config/env.config';
import './operators';

/**
 * This class represents the main application component.
 */
@Component({
  moduleId: module.id,
  selector: 'sd-app',
  templateUrl: 'app.component.html',
  styleUrls: ['app.component.css'],
})
export class AppComponent {
  constructor() {
    console.log('Environment config', Config);
  }
}
`

**Comment by vyakymenko** [comment]
@ravivit9 , if you will configure clean installation with my steps, all will work fine. Also, please learn MD text formatting.

**Comment by ravivit9** [comment]
Removed node_modules, did npm cahce clean --force, now installing all modules in package.json.
Sure will have a look MD text formatting.

**Comment by ravivit9** [comment]
Still the same error, give me sometime i will download angular-seed again in another directory and try to implement the above.

**Comment by ravivit9** [comment]
Downloaded angular-seed in another directory and tried the above steps which resulted in my original error screenshot, not sure what's missing.
<img width="888" alt="screen shot 2017-07-23 at 19 16 16" src="https://user-images.githubusercontent.com/3435829/28520098-1732199c-7066-11e7-955a-e1cb388b8d14.png">

**Comment by vyakymenko** [comment]
@ravivit9 , and you have done all steps?

**Comment by ravivit9** [comment]
Yes i have done all the above steps. the fresh app worked fine before doing the project.config.ts change afterwards the above error occurred.

1) Downloaded angular-seed
2) cd angular-seed
3) npm install
4) yarn install
5) npm start ( which worked fine as I can see the default output in browser)
---
6) Started performing the above suggested points and saved the project.
7) The above error occurred.

**Comment by vyakymenko** [comment]
@ravivit9 ,

<img width="840" alt="1" src="https://user-images.githubusercontent.com/7300673/28520692-3b5e438c-7079-11e7-82f6-bb7c76a05e50.png">
<img width="839" alt="2" src="https://user-images.githubusercontent.com/7300673/28520696-3e6ea454-7079-11e7-82eb-d703ec6a7a63.png">
<img width="838" alt="4" src="https://user-images.githubusercontent.com/7300673/28520698-4407b1f8-7079-11e7-91ae-d989ffd92fd6.png">
<img width="839" alt="5" src="https://user-images.githubusercontent.com/7300673/28520706-47f68d20-7079-11e7-9641-ef80a4362bdd.png">
<img width="840" alt="6" src="https://user-images.githubusercontent.com/7300673/28520707-4b32006e-7079-11e7-86a9-3602a742dae1.png">
<img width="840" alt="7" src="https://user-images.githubusercontent.com/7300673/28520711-4effe972-7079-11e7-83cf-39c8e5f4d1f2.png">
<img width="840" alt="10" src="https://user-images.githubusercontent.com/7300673/28520717-51e2932e-7079-11e7-9af2-abce0da2f0f1.png">
<img width="840" alt="11" src="https://user-images.githubusercontent.com/7300673/28520719-5624ede2-7079-11e7-9f3b-8abb326e1642.png">

**Comment by ravivit9** [comment]
I did exactly the same but only one difference.
I did npm install first then yarn install and then the following
npm install @angular/material @angular/cdk material-design-lite hammerjs --save

Will it make any difference?  If so I am happy to redo it once again.

**Comment by vyakymenko** [comment]
@ravivit9 , please look in screens and steps. Also, please learn about `yarn` and `npm`.

**Comment by ravivit9** [comment]
I again removed node_modules and cleared the npm cache with --force option.

Then npm install material-design-lite hammerjs --save

After which did a "npm install" and when i tried to run "npm start" it gave me an error so executed "yarn install" then "npm start" which worked fine.

Now I can see the website without any error.
<img width="308" alt="screen shot 2017-07-24 at 12 24 48" src="https://user-images.githubusercontent.com/3435829/28521349-20e6336a-706b-11e7-99ab-d065983731af.png">

Only thing is I wanted to use the angular/material2 instead of material-design-lite as I have a project already using angular/material2 but I wanted to use angular-seed as my primary base project.

Thank you so far @vyakymenko

**Comment by ravivit9** [comment]
Included material design2 js and css to the project.config.js and other changes as per the below screenshot.  Looks like the MD2 styling not applied.
<img width="308" alt="screen shot 2017-07-24 at 12 49 56" src="https://user-images.githubusercontent.com/3435829/28522190-eb73d558-706e-11e7-8422-e68a82f205b8.png">
<img width="583" alt="screen shot 2017-07-24 at 12 50 21" src="https://user-images.githubusercontent.com/3435829/28522192-efdd213a-706e-11e7-8b52-f3ca6601ab66.png">
<img width="817" alt="screen shot 2017-07-24 at 12 50 41" src="https://user-images.githubusercontent.com/3435829/28522198-f4d4a0a0-706e-11e7-9577-fedc64720b88.png">
<img width="349" alt="screen shot 2017-07-24 at 12 51 02" src="https://user-images.githubusercontent.com/3435829/28522199-f8b4eaa4-706e-11e7-87f6-0737b231ed57.png">

**Comment by vyakymenko** [comment]
@ravivit9 , it's not a seed issue, it's your own development issue.

**Comment by ravivit9** [comment]
ok.  Thanks for your involvement in sorting out this issue.  I will investigate further on this how to use MD2.

I am happy to close this ticket, if anything related to seed I will create a ticket.

**Comment by michelcve** [comment]
Syntax error, unexpected token '<' usually means some kind of path problem. Browser is loading HTML instead of javascript. To identify which file went wrong, simply look at the 'network' tab, and see which .js file spitted out html instead of javascript.

I also believe there was a bug with Zone.js 0.8.13 what resulted in something similar, but I'm not sure about that. But use the latest zone.js (0.8.16 at the time of writing) just to be sure ;-).

For your information, we are using Material Design 2 library with this project without issues, so it is possible.

**Comment by ravivit9** [comment]
Thanks for your update michelcve.  Will have a look and update you accordingly.

**Comment by ravivit9** [comment]
Just to update where I am now.

I have updated all npm libraries to make sure I have up to date libraries.

I believe there this.NPM_DEPENDENCIES is unable or not configured to find the route path in project.config.js.  When I included the following css file the final output not importing the css mentioned.
    this.NPM_DEPENDENCIES = [
      ...this.NPM_DEPENDENCIES,
      {src: '@angular/material/@angular/material.js', inject: 'libs'},
      {src: '@angular/material/prebuilt-themes/indigo-pink.css', inject: 'true'}
    ];

Then I included the following in the index file which loaded the indigo css property and styles appeared properly.
  <link href="node_modules/@angular/material/prebuilt-themes/indigo-pink.css" rel="stylesheet">

Similarly the @angular/material/@angular/material.js is still not loaded properly hence the token error in the console.  But in Network tab I don't find any error and material.js not being loaded in network tab.

After   <link href="node_modules/@angular/material/prebuilt-themes/indigo-pink.css" rel="stylesheet">

<img width="323" alt="screen shot 2017-08-04 at 10 48 45" src="https://user-images.githubusercontent.com/3435829/28964071-30a993be-7903-11e7-9730-22bb306edbd4.png">

<img width="512" alt="screen shot 2017-08-04 at 10 49 04" src="https://user-images.githubusercontent.com/3435829/28964084-40bad240-7903-11e7-81c1-93219b6de461.png">

**Comment by michelcve** [comment]
You should include the css like this:
      {src: '@angular/material/prebuilt-themes/indigo-pink.css', inject: true},

And replace the CSS from css/main.css with something like this (original css gets in the way for material's css):

body {
    font-family: Roboto,"Helvetica Neue",sans-serif;
    font-size: 16px;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    margin: 0;
    padding: 0;
}

**Comment by ravivit9** [comment]
Sorry, just noticed the path was truncated after inputting.
The config looks like this.
<img width="590" alt="screen shot 2017-08-04 at 12 36 12" src="https://user-images.githubusercontent.com/3435829/28967183-e24a59a6-7911-11e7-9bbc-c1c2baac8a7f.png">

**Comment by michelcve** [comment]
Can't you copy paste in a code block, instead of using images?

anyway, get rid of the
`{src: '@angular/material/@angular/material.js', inject: 'libs'}`

**Comment by michelcve** [comment]
And read this:

[https://github.com/mgechev/angular-seed/wiki/Integration-with-AngularMaterial2](https://github.com/mgechev/angular-seed/wiki/Integration-with-AngularMaterial2)

**Comment by ravivit9** [comment]
Thanks for your response.

Yes I did copied the actual code and after posting the comment part of the path disappeared.  That's the reason I took an image.  Infact I should have copied the code within the script quote.

`
    // Add `NPM` third-party libraries to be injected/bundled.
    this.NPM_DEPENDENCIES = [
      ...this.NPM_DEPENDENCIES,
      {src: '@angular/material/prebuilt-themes/indigo-pink.css', inject: 'true'}
    ];

    let additionalPackages: ExtendPackages[] = [
      {
        name: '@angular/material',
        path: 'node_modules/@angular/material/bundles/material.umd.js',
        packageMeta:{
          defaultExtension: 'js'
        }
        //path: '../../node_modules/@angular/material/@angular/material.js'
      },{
        name: '@angular/cdk',
        path: 'node_modules/@angular/cdk/bundles/cdk.umd.js',
        packageMeta:{
          defaultExtension: 'js'
        }
      },{
        name: 'hammerjs',
        path: 'node_modules/hammerjs/hammer.js',
        packageMeta:{
          defaultExtension: 'js'
        }
      }
    ];

    this.addPackagesBundles(additionalPackages);
`

After removing the material.js and I removed the css link from index file.  Then did a npm start to reload everything, this time no token error but the styling have gone which means the above NPM_DEPENDENCIES not taking effect.  Then added css file in index file again then it worked.

**Comment by ravivit9** [comment]
Pretty much deleted all entries and copied yours.

**Comment by michelcve** [comment]
Well, then I don't know. You must have an error somewhere else.

I'm using the CSS from the NPM_DEPENDENCIES, and it's working fine. And our application is using quite a few component from Material Design 2 I must say ;-)

**Comment by michelcve** [comment]
You could check you network tab, and see if it's loaded.

**Comment by ravivit9** [comment]
<img width="587" alt="screen shot 2017-08-04 at 14 13 16" src="https://user-images.githubusercontent.com/3435829/28970095-2233d544-791f-11e7-8ce8-c242d021093f.png">

<img width="477" alt="screen shot 2017-08-04 at 14 13 29" src="https://user-images.githubusercontent.com/3435829/28970101-28b34eb8-791f-11e7-864a-2b56530eae7a.png">

Would you mind sharing the piece of code in NPM_DEPENDENCIES?  If not, no issues.

**Comment by ravivit9** [comment]
Reason why I am after the NPM_DEPENDENCIES is because I would like to know the scope of it.  As I have another project which is MD2 quite ok.  I wanted to migrate my code base to angular-seed which I like very much.  I still can include the material css in index file but I need to be careful about future dependencies which may occur around NPM_DEPENDENCIES.

May be I will create a github temp seed with the changes and if you have time to have a look that would be great.

**Comment by michelcve** [comment]
This is my full NPM_DEPENDENCIES:

```
    // Add `NPM` third-party libraries to be injected/bundled.
    this.NPM_DEPENDENCIES = [
      ...this.NPM_DEPENDENCIES,
      /* Select a pre-built Material theme */
      {src: '@angular/material/prebuilt-themes/indigo-pink.css', inject: true},
    ];
```

If you look for a starter project, with this seed, and Material Design 2 included, [look here](https://github.com/Karasuni/angular-seed/commit/9993e924b909ec377cc1b59305bf697e83b38361) (it's not mine)

**Comment by michelcve** [comment]
And you should have had something like this in your network tab:

`indigo-pink.css?1501849246375`

So there is definitely something broken in your project. My suggestion? Create a new empty project, and follow there instructions [here](https://github.com/mgechev/angular-seed/wiki/Integration-with-AngularMaterial2). And also clean out the main.css as I mentioned before.

**Comment by ravivit9** [comment]
I have created brand new two angular-seed project after opening this ticket without any major change to the code base and followed the instructions as well.

Now I have created another new project with the link you suggested and giving it a try.  Will update once I have done that.

**Comment by ravivit9** [comment]
Did the following 
1) git clone from https://github.com/Karasuni/angular-seed
2) npm install
3) yarn install
4) npm start.

ALL WORKED FINE.
5) npm installed @angular/material, cdk, hammerjs
updated the project.config.js as per the guidelines

6) Added the following code in tools/shared.module.ts
`@NgModule({
  imports: [CommonModule, RouterModule, MdButtonModule
    ],
  declarations: [ToolbarComponent, NavbarComponent],
  exports: [ToolbarComponent, NavbarComponent,
    CommonModule, FormsModule, RouterModule,
    ]
})
export class SharedModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: SharedModule,
      providers: [NameListService]
    };
  }
}
`

 All I did was added MdButtonModule to the shared module file.  Then did a npm start to reload all the changes.  Now I can see the same console error.
<img width="909" alt="screen shot 2017-08-04 at 16 05 27" src="https://user-images.githubusercontent.com/3435829/28974647-f7cab272-792e-11e7-9897-3de3ea67a7fc.png">

No major changes to the app.  I have spent more than a week to figure it and and a good amount of time of other helpful group members as well.  But unfortunately I was not able to use this great seed in my project.

**Comment by ravivit9** [comment]
In fact even with the original angular-seed download I did changes only up to this much.

**Comment by mgechev** [comment]
For support requests you can use [gitter](https://gitter.im/mgechev/angular2-seed) or take a look at [project's the wiki page](https://github.com/mgechev/angular2-seed/wiki) in order to get more complete idea on how the it works.

**Comment by ravivit9** [comment]
I pretty much understood how the seed works and it is also configurable, but when things won't work as expected with no clue it is becoming bit of a hassle.  

I appreciate a whole community is working towards bringing this great seed to developers.  Probably me being unlucky as I tried in various different was to use angular-seed with MD2 including strictly following the instructions which is clear enough to understand and implement.

So far I have created in 5 different was to use in combination with MD2, no luck so far.  

At the moment I am using ng4-material seed but it doesn't have the advanced modularisation and configurable approach to it.  Hence thought of trying angular-seed.  But even a single MD2 module is not being imported and added to the app.

Hopefully someone may fork a angular-seed-MD2 version.  

Thanks a lot for all of your help.
